[docs] Adds AppJS banner

jonsamp opened the pull request:
# Why

I talked with Paulina about adding an App.js banner, to advertise it to Expo docs readers.

# How

- Adds the same component we had last year, but with updated information
- Adds the banner on the home page above the current content

# Screenshot

<img width="1348" alt="Screenshot 2023-03-08 at 11 31 11 AM" src="https://user-images.githubusercontent.com/6455018/223787161-f262d347-31f6-4f59-a1b5-2288e951e252.png">


# Test Plan

Make sure that the banner appears as it does in the screenshot above.

jonsamp committed:
Adds AppJS banner

jonsamp committed:
add date-fns

Simek reviewed (commented):
Left few small suggestions, also spacing between banner and home page title seems to be a bit too big, especially on mobile, but it might be just me. Overall looks good! 👍

jonsamp committed:
incorporate feedback

jonsamp committed:
convert to tailwind

jonsamp commented:
<img width="1744" alt="Screenshot 2023-03-13 at 11 32 28 AM" src="https://user-images.githubusercontent.com/6455018/224766121-08115ae3-d172-4624-a594-4ca422b5363e.png">

Here's an updated screenshot ^

Changes:
- I moved the banner to below the page title, since that makes more sense to me.
- I fixed a UI issue with the Quick Start gradient so that the gradient does not overflow the border on the top left and top right corners.
- Added the arrow up right icon to the Learn More button
- Converted the AppJSBanner to tailwind + use new components from styleguide

jonsamp committed:
delete unused imports

jonsamp committed:
lint

Simek reviewed (approved):
Please revert the changes to the `react-native` submodule before shipping this, otherwise LGTM, thank you for the updates! 👍

jonsamp committed:
reset react native lab
